core/vm/evm.go

```diff
@@ -227,10 +227,12 @@ func (self *EVM) Main(trx *Transaction) (ret ExecutionResult) {
 	gas_left -= gas_intrinsic
 
 	if contract_creation {
+		// setting nonce to current trx nonce to generate correct address for new contract. Nonce incremented inside of `create` later
```

Comment:
where is it incremented ? 

```
func (self *EVM) create_1(caller StateAccount, code []byte, gas uint64, value *big.Int) (ret []byte, contractAddr common.Address, leftOverGas uint64, err error) {
	contractAddr = crypto.CreateAddress(caller.Address(), caller.GetNonce())
	ret, leftOverGas, err = self.create(caller, CodeAndHash{Code: code}, gas, value, &contractAddr)
	return
}

func CreateAddress(b *common.Address, nonce uint64) (ret common.Address) {
	copy(ret[:], keccak256.Hash(rlp.MustEncodeToBytes([2]interface{}{b, nonce}))[12:])
	return
}
```

Reply:
https://github.com/Taraxa-project/taraxa-evm/blob/befcbc4e2c3394c9476298c85fc5118bd663303c/core/vm/evm.go#L284